LoopUI/Views/SettingsViewModel.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  SettingsViewModel.swift
+//  LoopUI
+//
+//  Created by Rick Pasetto on 6/25/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Combine
+import LoopCore
+import LoopKit
+import LoopKitUI
+import SwiftUI
+
+public class DeviceViewModel: ObservableObject {
+    public init(deviceManagerUI: DeviceManagerUI.Type? = nil,
+                isSetUp: Bool = true,
+                onTapped: @escaping () -> Void = { }) {
+        self.deviceManagerUI = deviceManagerUI
+        self.isSetUp = isSetUp
+        self.onTapped = onTapped
+    }
+    
+    let deviceManagerUI: DeviceManagerUI.Type?
+
+    @Published private(set) var isSetUp: Bool = false
+    
+    var image: UIImage? { deviceManagerUI?.smallImage }
+    // TODO: Remove the defaults here...they are only here for illustrative purposes
+    var name: String { deviceManagerUI?.name ?? "device" }
+    var details: String { deviceManagerUI?.details ?? "device details"  }
+   
+    let onTapped: () -> Void
+}
+
+public class SettingsViewModel: ObservableObject {
+    
+    let notificationsCriticalAlertPermissionsViewModel: NotificationsCriticalAlertPermissionsViewModel
+
+    @Published var appNameAndVersion: String
+    @Published var dosingEnabled: Bool = false {
```

Comment:
Looks like the initial value comes from the initializer—I don't think you need a default value here